utils/jwt_utils.py

```
@@ -0,0 +1,88 @@
+        payload = json.loads(_b64url_decode(payload_part).decode("utf-8"))
+        if int(payload["exp"]) <= int(datetime.now(timezone.utc).timestamp()):
+            raise _unauthorized()
+
+        return {
+            "user_id": int(payload["sub"]),
+            "token_version": int(payload.get("ver", 0)),
+        }
+    except HTTPException:
+        raise
+    except (ValueError, KeyError, TypeError, json.JSONDecodeError, base64.binascii.Error):
+        raise _unauthorized()
```

Ruff (0.15.6)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)